_source/_posts/2022-06-14-simple-crud-react-and-spring-boot.md

```diff
@@ -0,0 +1,1348 @@
+import './App.css';
+
+const App = () => {
+
+  const [groups, setGroups] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+
+    fetch('api/groups')
+      .then(response => response.json())
+      .then(data => {
+        setGroups(data);
+        setLoading(false);
+      })
+  }, [setGroups, setLoading]);
+
+  if (loading) {
+    return <p>Loading...</p>;
+  }
+
+  return (
+    <div className="App">
+      <header className="App-header">
+        <img src={logo} className="App-logo" alt="logo" />
+        <div className="App-intro">
+          <h2>JUG List</h2>
+          {groups.map(group =>
+            <div key={group.id}>
+              {group.name}
+            </div>
+          )}
+        </div>
+      </header>
+    </div>
+  );
+}
+
+export default App;
+```
+
+To proxy from `/api` to `http://localhost:8080/api`, add a proxy setting to `app/package.json`.
+
+```json
+"scripts": {...},
+"proxy": "http://localhost:8080",
+```
+
+To learn more about proxying API requests, see Create React App's [documentation](https://create-react-app.dev/docs/proxying-api-requests-in-development/#docsNav). 
+
+Make sure Spring Boot is running, then run `npm start` in your `app` directory. You should see the list of default groups. 
+
+{% img blog/spring-boot-react/jug-list.png alt:"JUG List" width:"800" %}{: .center-image }
+
+## Build a React `GroupList` component
+
+React is all about components, and you don't want to render everything in your main `App`, so create `app/src/GroupList.js` and populate it with the following JavaScript.
+
+{% raw %}
+```jsx
+import React, { useEffect, useState } from 'react';
+import { Button, ButtonGroup, Container, Table } from 'reactstrap';
+import AppNavbar from './AppNavbar';
+import { Link } from 'react-router-dom';
+
+const GroupList = () => {
+
+  const [groups, setGroups] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+
+    fetch('api/groups')
+      .then(response => response.json())
+      .then(data => {
+        setGroups(data);
+        setLoading(false);
+      })
+  }, [setGroups, setLoading]);
+
+  const remove = async (id) => {
+    await fetch(`/api/group/${id}`, {
+      method: 'DELETE',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      }
+    }).then(() => {
+      let updatedGroups = [...groups].filter(i => i.id !== id);
+      setGroups(updatedGroups);
+    });
+  }
+
+  if (loading) {
+    return <p>Loading...</p>;
+  }
+
+  const groupList = groups.map(group => {
+    const address = `${group.address || ''} ${group.city || ''} ${group.stateOrProvince || ''}`;
+    return <tr key={group.id}>
+      <td style={{whiteSpace: 'nowrap'}}>{group.name}</td>
+      <td>{address}</td>
+      <td>{group.events.map(event => {
+        return <div key={event.id}>{new Intl.DateTimeFormat('en-US', {
```

Comment:
return seems unnecessary as this can be pure lambda

Reply:
IntelliJ gives me a warning if I remove it.
<img width="925" alt="Screen Shot 2022-06-16 at 18 26 13" src="https://user-images.githubusercontent.com/17892/174199068-39ae6bea-6efa-4fbc-a7f0-43060642dd26.png">

Reply:
cox the `{` needs to be changed to `(` I think